src/sqlfluff/core/config.py

```diff
@@ -1211,16 +1239,14 @@ def process_inline_config(self, config_line: str, fname: str):
             )
             return
         config_line = config_line[9:].strip()
-        # Divide on colons
-        config_path = [elem.strip() for elem in config_line.split(":")]
-        config_val = (tuple(config_path[:-1]), config_path[-1])
+        config_val = split_colon_separated_string(config_line)
         # Validate the value
         ConfigLoader._validate_configs([config_val], fname)
         # Set the value
         self.set_value(*config_val)
         # If the config is for dialect, initialise the dialect
-        if config_path[:-1] == ["dialect"]:
-            self._initialise_dialect(config_path[-1])
+        if config_val[0] == ("dialect",):
```

Comment:
```suggestion
        if config_val[0] == "dialect":
```

Reply:
Unfortunately this one _does_ need to be a `tuple`. I got it wrong the first time and it caused a weird bug. I'll add a comment to explain.